app/services/search/postgres/reading_list.rb

```diff
@@ -58,12 +59,20 @@ def self.search_documents(user, term: nil, statuses: [], tags: [], page: 0, per_
         }
       end
 
-      def self.find_articles(user_id:, term:, statuses:, tags:, page:, per_page:)
-        relation = ::Article
-          .joins(:reactions)
-          .where("reactions.category": :readinglist)
-          .where("reactions.user_id": user_id)
-          .where("reactions.status": statuses)
+      def self.find_articles(user:, term:, statuses:, tags:, page:, per_page:)
+        # [@jgaskins, @rhymes] as `reactions` is potentially a big table, adding pagination
+        # to an INNER JOIN (eg. `joins(:reactions)`) exponentially decreases the performance,
+        # incrementing query time as the database has to scan all the rows just to discard
+        # them right after if they lie outside the bounds of the `OFFSET`.
```

Comment:
I'm fine with the current solution but I often share this post on `OFFSET` based pagination:

https://use-the-index-luke.com/sql/partial-results/fetch-next-page

Reply:
Yep, we could totally implement different pagination strategies, but that would require changing quite a bit of code and I'm not sure we should do that now

Reply:
No, we absolutely shouldn't do this now. Sorry if I was unclear on that. I just often share this article because `OFFSET` based pagination has a lot of issues, but despite that it's commonly used. If we ever were to change this it'd certainly be in a different PR.